master/internal/rm/kubernetesrm/pods.go

```diff
@@ -1291,15 +1295,23 @@ func (p *pods) handleGetSlotRequest(agentID string, slotID string) *apiv1.GetSlo
 }
 
 func (p *pods) handleGetAgentsRequest() *apiv1.GetAgentsResponse {
-	nodeSummaries := p.summarizeClusterByNodes()
-	_, nodesToPools := p.getNodeResourcePoolMapping(nodeSummaries)
+	p.getAgentsCacheLock.Lock()
+	defer p.getAgentsCacheLock.Unlock()
+
+	if time.Since(p.getAgentsCacheTime) > 5*time.Second {
```

Comment:
Why 5s for a cache timeout? That seems pretty short. 

Nit: I would prefer that the cache timeout be called out as a constant instead of hard-coded.

Reply:
5 seconds was chosen because it is the same as the summarize cache. The cache was made to resolve a performance issue so I think we have pretty good evidence that 5 seconds seems reasonable to prevent performance issues. The longer the cache the less response the ui / cli is. I made it a constant.

What time would you choose?

Reply:
If 5 seconds works in practice already, that works for me! Does the API have a timeout? 

I'm concerned about a case where (picking numbers out of the air) there's an API call every 1s, with a cache timeout of 5s. But the total time to update the cache is 15s. The cache is locked for reads and writes while it's being updated right? That could result in a lot of stalled API calls. And by the time the cache is updated, it's out of date again.

I'm assuming it's unlikely that the call to `listPodsInAllNamespaces` would take a long time ... but I think it's possible since it involves a call to k8s (right?).